lib/auth/default_current_user_provider.rb

```diff
@@ -378,27 +388,53 @@ def header_api_key?
     !!@env[HEADER_API_KEY]
   end
 
-  def rate_limit_admin_api_requests!
-    return if Rails.env == "profile"
+  def can_write?
+    @can_write ||= !Discourse.pg_readonly_mode?
+  end
+
+  def admin_api_key_limiter
+    return @admin_api_key_limiter if @admin_api_key_limiter
 
     limit = GlobalSetting.max_admin_api_reqs_per_minute.to_i
     if GlobalSetting.respond_to?(:max_admin_api_reqs_per_key_per_minute)
       Discourse.deprecate("DISCOURSE_MAX_ADMIN_API_REQS_PER_KEY_PER_MINUTE is deprecated. Please use DISCOURSE_MAX_ADMIN_API_REQS_PER_MINUTE")
-      limit = [ GlobalSetting.max_admin_api_reqs_per_key_per_minute.to_i, limit].max
+      limit = [
+        GlobalSetting.max_admin_api_reqs_per_key_per_minute.to_i,
+        limit
+      ].max
     end
-
-    global_limit = RateLimiter.new(
+    @admin_api_key_limiter = RateLimiter.new(
       nil,
       "admin_api_min",
       limit,
-      60
+      60,
+      error_code: "admin_api_key_rate_limit"
     )
-
-    global_limit.performed!
   end
 
-  def can_write?
-    @can_write ||= !Discourse.pg_readonly_mode?
+  def user_api_key_limiter_60_secs
+    return @user_api_key_limiter_60_secs if @user_api_key_limiter_60_secs
```

Comment:
Just curious, it seems like we only use this once so we technically don't have to memoize this?